### DISCUSSION TITLE
Addressing the reinit of a model from a previous state that includes delay() functions

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#13431] [Q&A]
Hi, 

Is it possible to save and restore the history associated with the delay function so that the simulation of a model can be properly restored from a snapshot? 

Thanks in advance!

**Comment by casella** [comment]
Not at the moment, but we could think about it

**Reply by mpradovelasco** [reply]
It would be really interesting to have this capability in pharmacometric and biosystems models, where the delay is widely used to calculate half-life, area under the curve, and other standard indices that support prescription optimization and genomic and personalized precision medicine. 
Its implementation on openModelica will add a new issue where OM overcomes other object oriented Modeling and Simulation development environments like EcosimPro :)

**Comment by casella** [comment]
@mpradovelasco thanks for reporting this user requirement.

This feature is related to being able to initialize the delay state. This is currently possible with the [spatialDistribution()](https://specification.modelica.org/maint/3.6/operators-and-expressions.html#modelica:spatialDistribution) operator, but it isn't with the simpler [delay()](https://specification.modelica.org/maint/3.6/operators-and-expressions.html#modelica:delay) operator.

I think before implementing such a feature we should have a discussion at the level of the Modelica Language Specification.

**Reply by casella** [reply]
See modelica/ModelicaSpecification#3633

**Reply by mpradovelasco** [reply]
Thanks a lot for considering this feature.
